modules/thermal_hydraulics/src/physics/THMVACESinglePhaseFlowPhysics.C

```diff
@@ -0,0 +1,662 @@
+      _sim->addAuxKernel(class_name, genName(comp_name, "v_aux"), params);
+    }
+    {
+      std::string class_name = "THMSpecificInternalEnergyAux";
+      InputParameters params = _factory.getValidParams(class_name);
+      params.set<AuxVariableName>("variable") = SPECIFIC_INTERNAL_ENERGY;
+      params.set<std::vector<SubdomainName>>("block") = flow_channel->getSubdomainNames();
+      params.set<std::vector<VariableName>>("rhoA") = {RHOA};
+      params.set<std::vector<VariableName>>("rhouA") = {RHOUA};
+      params.set<std::vector<VariableName>>("rhoEA") = {RHOEA};
+      _sim->addAuxKernel(class_name, genName(comp_name, "e_aux"), params);
+    }
+
+    {
+      std::string class_name = "PressureAux";
+      InputParameters params = _factory.getValidParams(class_name);
+      params.set<AuxVariableName>("variable") = PRESSURE;
+      params.set<std::vector<SubdomainName>>("block") = flow_channel->getSubdomainNames();
+      params.set<std::vector<VariableName>>("e") = {SPECIFIC_INTERNAL_ENERGY};
+      params.set<std::vector<VariableName>>("v") = {SPECIFIC_VOLUME};
+      params.set<UserObjectName>("fp") = _fp_name;
+      _sim->addAuxKernel(class_name, genName(comp_name, "pressure_uv_auxkernel"), params);
+    }
+    {
+      std::string class_name = "TemperatureAux";
+      InputParameters params = _factory.getValidParams(class_name);
+      params.set<AuxVariableName>("variable") = TEMPERATURE;
+      params.set<std::vector<SubdomainName>>("block") = flow_channel->getSubdomainNames();
+      params.set<std::vector<VariableName>>("e") = {SPECIFIC_INTERNAL_ENERGY};
+      params.set<std::vector<VariableName>>("v") = {SPECIFIC_VOLUME};
+      params.set<UserObjectName>("fp") = _fp_name;
+      _sim->addAuxKernel(class_name, genName(comp_name, "T_auxkernel"), params);
+    }
+
+    {
+      std::string class_name = "SpecificTotalEnthalpyAux";
+      InputParameters params = _factory.getValidParams(class_name);
+      params.set<AuxVariableName>("variable") = SPECIFIC_TOTAL_ENTHALPY;
+      params.set<std::vector<SubdomainName>>("block") = flow_channel->getSubdomainNames();
+      params.set<std::vector<VariableName>>("rhoA") = {RHOA};
+      params.set<std::vector<VariableName>>("rhoEA") = {RHOEA};
+      params.set<std::vector<VariableName>>("p") = {PRESSURE};
+      params.set<std::vector<VariableName>>("A") = {AREA};
+      _sim->addAuxKernel(class_name, genName(comp_name, "H_auxkernel"), params);
+    }
+  }
+}
+
+void
+THMVACESinglePhaseFlowPhysics::addUserObjects()
+{
+  const std::string class_name = "ADNumericalFlux3EqnHLLC";
+  InputParameters params = _factory.getValidParams(class_name);
+  params.set<UserObjectName>("fluid_properties") = _fp_name;
+  params.set<MooseEnum>("emit_on_nan") = "none";
+  params.set<ExecFlagEnum>("execute_on") = {EXEC_INITIAL, EXEC_LINEAR, EXEC_NONLINEAR};
+  _sim->addUserObject(class_name, _numerical_flux_name, params);
+}
+
+void
+THMVACESinglePhaseFlowPhysics::addInletBoundaries()
```

Comment:
Do we need to distinguish inlet/outlet, or can we just have a generic `addFlowBCs()` or whatever?

Reply:
modeling after WCNSFV here. in wcnsfv we do need to distinguish